File: artemis-entity-tracker/src/main/java/net/namekdev/entity_tracker/utils/tuple/Tuple4.java
package net.namekdev.entity_tracker.utils.tuple;

public final class Tuple4<E1, E2, E3, E4> {
	public E1 item1;
	public E2 item2;
	public E3 item3;
	public E4 item4;

	public Tuple4(E1 item1, E2 item2, E3 item3, E4 item4) {
		this.item1 = item1;
		this.item2 = item2;
		this.item3 = item3;
		this.item4 = item4;
	}

	public static <E1, E2, E3, E4> Tuple4<E1, E2, E3, E4> create(E1 item1, E2 item2, E3 item3, E4 item4) {
		return new Tuple4<E1, E2, E3, E4>(item1, item2, item3, item4);
	}
}

File: src/commonMain/kotlin/net/namekdev/entity_tracker/utils/tuple/Tuple4.kt
package net.namekdev.entity_tracker.utils.tuple

class Tuple4<E1, E2, E3, E4>(var item1: E1, var item2: E2, var item3: E3, var item4: E4) {
    companion object {

        fun <E1, E2, E3, E4> create(item1: E1, item2: E2, item3: E3, item4: E4): Tuple4<E1, E2, E3, E4> {
            return Tuple4(item1, item2, item3, item4)
        }
    }
}
